lib/miq_automation_engine/engine/miq_ae_builtin_method.rb

```diff
@@ -201,13 +201,46 @@ def self.detect_vendor(src_obj, attr)
       return unless src_obj
       case attr
       when "orchestration_stack"
-        src_obj.type.split('::')[2]
+        src_obj.ext_management_system.try(:provider_name)
       when "miq_host_provision"
         "vmware"
       when "miq_request", "miq_provision", "vm_migrate_task"
-        src_obj.source.try(:vendor)
+        src_obj.source.try(:provider_name)
       when "vm"
-        src_obj.try(:vendor)
+        src_obj.try(:provider_name)
+      end
+    end
+
+    def self.emsevent_provider_name(event_stream)
+      return nil if event_stream.ext_management_system.nil?
+      event_stream.ext_management_system.try(:provider_name)
+    end
+    private_class_method :emsevent_provider_name
+
+    def self.emsevent_manager_type(event_stream)
+      return nil if event_stream.ext_management_system.nil?
+      manager_type = event_stream.ext_management_system.try(:manager_type).downcase
+      manager_type.casecmp("infra").zero? ? INFRASTRUCTURE : manager_type
```

Comment:
@tinaafitz 
Since we are lady downcased in the previous line can we just check if manager_type == "infra"